### PR TITLE
✨  Rosa machinepools

### PR DESCRIPTION
**Pull request by enxebre**
**What type of PR is this?**
Add initial support for ROSAMachinePools

Includes https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4665

**What this PR does / why we need it**:
This PR introduces basic support to create/delete ROSAMachinePools
It also add several improvements to the rosa control plane to get cluster creation functional:
    - Add ROSAControlPlaneReadyCondition
    - Add helpers for ocmclient
    - Add WorkerRoleARN field to the API to satisfy latest ocm API requirements

Sample:
```
---
apiVersion: cluster.x-k8s.io/v1beta1
kind: MachinePool
metadata:
  name: replace-me
  namespace: replace-me
spec:
  clusterName: replace-me
  replicas: 2
  template:
    spec:
      clusterName: replace-me
      bootstrap:
        dataSecretName: ""
      infrastructureRef:
        apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
        kind: ROSAMachinePool
        name: replace-me
---
apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
kind: ROSAMachinePool
metadata:
  name: replace-me
  namespace: replace-me
spec:
  aws:
    instanceType: "m5.xlarge"
    subnet: "replace-me"
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4429

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Introduce basic support for ROSAMachinePools

```

### CODE REVIEW

**Comment by enxebre** [comment]
cc @vincepri @muraee

File: controlplane/rosa/controllers/rosacontrolplane_controller.go [hunk 1]
```diff
@@ -417,13 +407,66 @@ func (r *ROSAControlPlaneReconciler) rosaClusterToROSAControlPlane(log *logger.L
...
+		return nil, handleErr(cluster.Error(), err)
+	}
+
+	clusterObject := cluster.Body()
+
+	return clusterObject, nil
+}
+
+func (client *OCMClient) GetCluster(rosaScope *scope.ROSAControlPlaneScope) (*cmv1.Cluster, error) {
+	clusterKey := rosaScope.ControlPlane.Name[:15]
```
Comment (vincepri) [inline controlplane/rosa/controllers/rosacontrolplane_controller.go:464]: Same as above, could we do this somewhere else?

File: controlplane/rosa/controllers/rosacontrolplane_controller.go [hunk 2]
```diff
@@ -417,13 +407,66 @@ func (r *ROSAControlPlaneReconciler) rosaClusterToROSAControlPlane(log *logger.L
 	}
 }
 
-func (r *ROSAControlPlaneReconciler) getOcmCluster(rosaScope *scope.ROSAControlPlaneScope, ocmConnection *sdk.Connection) (*cmv1.Cluster, error) {
-	clusterKey := rosaScope.ControlPlane.Name
+// OCMClient is a temporary helper to talk to OCM API.
+// TODO(alberto): vendor this from https://github.com/openshift/rosa/tree/master/pkg/ocm or build its own package here.
```
Comment (vincepri) [inline controlplane/rosa/controllers/rosacontrolplane_controller.go:411]: Is this vendorable? Why have we decided not to import it today?

Comment (muraee) [reply controlplane/rosa/controllers/rosacontrolplane_controller.go:411]: We would have to vendor the complete rosa cli repo https://github.com/openshift/rosa, I don't think it is possible to only import the ocm package

**Pull request closed without merging**